sdk/language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/ClassGenUtils.scala

```diff
@@ -69,11 +83,24 @@ private[inner] object ClassGenUtils {
       .initializer(
         "new $T($S, $S, $S)",
         classOf[javaapi.data.Identifier],
-        packageId,
+        packageRef,
```

Comment:
I've sync'ed with @remyhaemmerle-da offline and this is the way to go for 2.9, thank you

Reply:
If the user wants to explicitly target only the current package ID, they could also use

```
new Ref.Identifier(Foo.PACKAGE_ID, Foo.TEMPLATE_ID.getModuleName(), Foo.TEMPLATE_ID.getEntityName())
```

It's awkward, but perhaps appropriately so, as this is not a default usage.

Reply:
I've just realized something after you mentioned deactivating the conformance tests which I wouldn't be comfortable to do before the release. I've looked at them and it seems to me that we might be able to satisfy the (as I understand them) requirements of this PR with a slightly different approach.

As far as In understand, we want to switch the default interaction method of the Java bindings to the Ledger API from using the template-id FQN to a template-reference based on the package-name selector (of form `#<package-name>:<mod>:<entity-name>`. So the easiest way to do was to indeed change the `TEMPLATE_ID` as proposed here. But there are some caveats to this so how about we do as follows:
* Use the `PACKAGE_NAME` field that I've introduced a couple of days ago (and you backported afaik @raphael-speyer-da )
* Introduce a `private static final Identifier TEMPLATE_REF = new Identifier(s"#$PACKAGE_NAME", <mod>, "<entity>");` 
* Replace everywhere in the codegen where we use the `TEMPLATE_ID` with the `TEMPLATE_REF`. Basically I think this is just in commands and choices. I see only three uses of this field name in the codegen builder code so it might be simple. 

This allows us to achieve the initial goal of this PR with minimal impact. Then we are free later to:
* Think of a way to make this behavior toggleable (switch dynamically between `TEMPLATE_REF` and `TEMPLATE_ID` in commands)
* Adapt conformance tests to use by-package-name queries/submissions

What do you think @raphael-speyer-da ?